src/game/AI/CreatureAI.cpp

```diff
@@ -142,7 +142,9 @@ CanCastResult CreatureAI::DoCastSpellIfCan(Unit* pTarget, uint32 uiSpell, uint32
             // Creature should stop wielding weapon while casting
             pCaster->SetSheath(SHEATH_STATE_UNARMED);
 
-            pCaster->CastSpell(pTarget, pSpell, !!(uiCastFlags & CAST_TRIGGERED), nullptr, nullptr, uiOriginalCasterGUID);
+            uint32 flags = (uiCastFlags & CAST_TRIGGERED ? TRIGGERED_OLD_TRIGGERED : TRIGGERED_NONE) | (uiCastFlags & CAST_IGNORE_UNSELECTABLE_TARGET ? TRIGGERED_IGNORE_UNSELECTABLE_FLAG : TRIGGERED_NONE);
```

Comment:
This right here goes to show it's never been about readability but rather about hating on me and my code in the past. !(A || B) is hard to grasp but this right here is clear as day,. eh?